uwmweb/uwmbase
# hotfix/uml-dropdown-position > develop

QA issues:
- https://www.wrike.com/open.htm?id=336634172 - position of dropdown on focus vs click toggle
  - (FIXED/validated on stage)
- https://www.wrike.com/open.htm?id=336629827 - extraneous `_this.ShowLocation` in code 
  - (FIXED/validated on stage)
- https://www.wrike.com/open.htm?id=336655635 - UML does not work on Safari, IE
  - TO TEST on stage
- https://www.wrike.com/open.htm?id=336682387 - hitting enter to search sometimes does not use current location
  - TO TEST on stage?

CMS PR: https://github.com/uwmweb/uwmcms/pull/785

This tweaks sass and js for the use-my-location dropdown toggling.

* Don't use Bootstrap dropdown component, as it was causing undesired styling issues
* On click of current-location arrow icon, toggle show/hide the dropdown
* Open dropdown:
 * on focus of address input
* Close dropdown:
  * if focus moves to a different element than the address input or dropdown
  * on click of Use-my-location link, after triggering browser request
  * when geocoding of user-typed text completes
* In the JS - selecting `$form` now uses the `context` variable to ensure it only happens on the correct Drupal attach behaviors call (avoiding attaching event handlers multiple times) - this required not using `$form` in the helper functions, so I simplified selectors for the inputs there
* In JS, moved geolocation helper functions inside `attach()` so they can use the jquery-selected element variables already defined
* Updated styling to apply the current-location purple icon via the conditional body class (this bit was lost with changing to the custom SVG)
* Changed `uml` hidden input to `latlng` because "uml" was becoming an umlaut character in the URL on IE
